src/test/java/org/mockitousage/matchers/CapturingArgumentsTest.java

```diff
@@ -298,7 +313,7 @@ public void captures_correctly_when_captor_used_multiple_times() throws Exceptio
 
         // then
         // this is only for backwards compatibility. It does not make sense in real to do so.
-        verify(mock).mixedVarargs(any(), argumentCaptor.capture(), argumentCaptor.capture(), argumentCaptor.capture());
+        verify(mock).mixedVarargs(any(), argumentCaptor.capture());
```

Comment:
@terebesirobert Actually I don't understand what causes the failure. I am looking forward to your solution! 
About adding equals/hashCode to each matcher is a bit dangerous. Imagine a programmer provides customer matcher but simply don't know  that eq/hashCode must be implemented. Than other team member use this custom matcher with success in the most common cases but in the above described scenario they get an failure and don't know why. 
Java 8 Lambda expression are also not possible (`captureIf(String s-> s.length()==5)`) cause they can't implement eq/hashCode.